### ISSUE TITLE
DefaultLocalPersistenceService.recursiveDeleteDirectoryContent(File) always returns true

### ISSUE THREAD

**Issue by mathieucarbou** [ehcache/ehcache3#1161]
This method is used in `DefaultLocalPersistenceService.destroyAll()`, which then always records a "positive" action _"Destroyed all file based persistence contexts"_

Content of this method is:

```
  private static boolean recursiveDeleteDirectoryContent(File file) {
    File[] contents = file.listFiles();
    if (contents == null) {
      throw new IllegalArgumentException("File " + file.getAbsolutePath() + " is not a directory");
    } else {
      boolean deleteSuccessful = true;
      for (File f : contents) {
        deleteSuccessful |= tryRecursiveDelete(f);
      }
      return deleteSuccessful;
    }
  }
```

I guess probably a fix could be:

```
  private static boolean recursiveDeleteDirectoryContent(File file) {
    File[] contents = file.listFiles();
    if (contents == null) {
      throw new IllegalArgumentException("File " + file.getAbsolutePath() + " is not a directory");
    } if (contents.length == 0) {
      return true;
    } else {
      boolean deleteSuccessful = false;
      for (File f : contents) {
        deleteSuccessful |= tryRecursiveDelete(f);
      }
      return deleteSuccessful;
    }
  }
```

Let me know if you are ok with this fix - I can send a PR.

**Comment by chrisdennis** [comment]
A better fix would be:
```java
private static boolean recursiveDeleteDirectoryContent(File file) {
  File[] contents = file.listFiles();
  if (contents == null) {
    throw new IllegalArgumentException("File " + file.getAbsolutePath() + " is not a directory");
  } else {
    boolean deleteSuccessful = true;
    for (File f : contents) {
      deleteSuccessful &= tryRecursiveDelete(f);
    }
    return deleteSuccessful;
  }
}
```

**Comment by mathieucarbou** [comment]
ok! Returns true only if all succeeded.

**Comment by ljacomet** [comment]
Re-opening as this needs to be fixed on `release/3.0` as well.
